docs/practices/appeng/adrs/0005-javascript-dependency-management.mdx

```diff
@@ -0,0 +1,273 @@
+---
+title: 0005 - JavaScript dependency management
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# 0005 - JavaScript dependency management
+
+**Status:** WIP (_initial draft_)
+
+**Date Accepted:** n/a for now
+
+**Reviewers:** #g-frontend
+
+:::info
+This ADR uses Tabs to separate specific code or documentation between the
+`npm-cli` and the `classic yarn-cli`.
+:::
+
+## Context
+
+Truss leverages both _NPM_ and _Yarn_ for dependency management for JavaScript
+projects. While these tools accomplish similar tasks, there are different in
+ways they operate. Teams who choose one tool over the other, must have
+documentation related to managing dependencies specific for their that team.
+
+Truss can provide best practices for how to manage JavaScript dependencies that
+can be used as a starting point for a team's or project's specific tooling
+around dependency management. This ADR seeks to do just that.
+
+The specific areas this ADR covers are the following:
+
+- dependency resolution
+    - version locking
+- updating dependencies
+- working with files such as: `package.json`, `package-lock.json`, and
+  `yarn.lock`.
+
+### JavaScript Dependency Management Tool
+
+Truss recommends that projects use _NPM_ instead of _Yarn_ as it is included with
+Node and has commercial support. Projects that leverage _Yarn_ or any other
+**JavaScript Dependency Management Tool** (JDMT from now on) will need to
+document this in ways outside of the scope of this ADR.
+
+This ADR is not meant to endorse one JDMT over the other and ultimately places the
+decision on engineering teams using these tools. There are certain ways these
+tools work together and this document will aim to cover those scenarios
+equitably.
```

Comment:
I'm not too sure about this section here. I think it might be good to establish use of _NPM_ over _Yarn_ for Truss projects, but I also have not been on any Truss projects that use _NPM_ over _Yarn_. This is why this opening paragraph is worded so weirdly. 

[=> Slack thread on this topic](https://trussworks.slack.com/archives/CD7MJLFJ4/p1654609942295709).